File: Swift3/AlgorithmExercise/HackerRank.swift
//
//  HackerRank.swift
//  AlgorithmExercise
//
//  Created by 이미정 on 2017. 11. 11..
//  Copyright © 2017년 이미정. All rights reserved.
//

import Foundation

class HackerRank {
    class WarmUp {
        static func ComparetheTriplets() {
            let aPoints = readLine()!.split(separator: " ").map{ Int($0)! }
            let bPoints = readLine()!.split(separator: " ").map{ Int($0)! }
            
            var alice = 0; var bob = 0;
            for i in 0..<aPoints.count {
                
                if aPoints[i] > bPoints[i] {
                    alice += 1
                }else if aPoints[i] < bPoints[i] {
                    bob += 1
                }
            }
            print("\(alice) \(bob)")
        }
        
        static func DiagonalDifference() {
            // read the integer n
            let n = Int(readLine()!)!
            
            // declare 2d array
            var arr: [[Int]] = []
            
            // read array row-by-row
            for _ in 0..<n {
                arr.append(readLine()!.components(separatedBy: " ").map{ Int($0)! })
            }
            
            var sum1 = 0; var sum2 = 0
            for i in 0..<n {
                sum1 += arr[i][i]
                sum2 += arr[i][n - i - 1]
            }
            
            print(abs(sum1 - sum2))
        }
    }
}

File: Java/src/hackerRank/HackerRank.java
package hackerRank;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class HackerRank {
    public static void main(String[] args) throws IOException {
        WarmUp.DiagonalDifference();
    }

    static class WarmUp {
        static void DiagonalDifference() throws IOException {
            InputStreamReader isr = new InputStreamReader(System.in);
            BufferedReader br = new BufferedReader(isr);

            int count = Integer.parseInt(br.readLine());
            String[][] input = new String[count][count];
            int sum1 = 0, sum2 = 0;

            for(int i=0; i<count; i++) {
                input[i] = br.readLine().split(" ");
                sum1 += Integer.parseInt(input[i][i]);
                sum2 += Integer.parseInt(input[i][count - i - 1]);
            }
            System.out.print(Math.abs(sum1 - sum2));
        }
    }
}
